src/display/pattern_helper.js

```diff
@@ -322,63 +322,58 @@ var TilingPattern = (function TilingPatternClosure() {
 
       var x0 = bbox[0], y0 = bbox[1], x1 = bbox[2], y1 = bbox[3];
 
-      var topLeft = [x0, y0];
-      // we want the canvas to be as large as the step size
-      var botRight = [x0 + xstep, y0 + ystep];
+      // It is a waste of memory to use the size as specified by xstep or ystep
+      // if the bounding box is smaller.
+      xstep = Math.min(xstep, x1);
```

Comment:
See also http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#page=183&zoom=180,23,420 

Also notice XStep/YStep comments below:
- "XStep  and  YStep  may  differ  from  the  dimensions  of  the  pattern  cell  implied  by  the  BBox  entry."
- "XStep and YStep may be either positive or negative but shall not be zero. "